### PR TITLE
addrs: Simplify presentation of provider FQNs

### PR DESCRIPTION
**Pull request by alisdair**
The provider fully-qualified name string used in configuration is very long, and since most providers are hosted in the public registry, most of that length is redundant. This commit adds and uses a `ForDisplay` method, which simplifies the presentation of provider FQNs.

If the hostname is the default hostname, we now display only the namespace and type. This is only used in UI, but should still be unambiguous, as it matches the FQN string parsing behaviour.

**Note:** I've probably have missed some places where we output provider strings intended for users to read. Suggestions on how to find them (or pointers to places I've missed) would be great 🙏 

### Screenshots

Before:
before

After:
after

### CODE REVIEW

**Comment by mildwonkey** [comment]
This is great!

**Comment by mildwonkey** [comment]
This is great! I have a couple of thoughts:

What do you think about renaming the function to something a bit more descriptive, like `ForDisplay()` or `DisplayString()`? 

I think the error messages/diagnostics should always use the full FQN; that's exactly where we need to avoid ambiguity. What do you think?

**Comment by alisdair** [comment]
Both good points, thanks! I went with `ForDisplay`, since svchost's `ForDisplay` method is used in proximity. And constraining the use (for now) to the init command log hooks makes the diff much smaller 😀

File: addrs/provider.go [hunk 1]
```diff
@@ -44,14 +44,27 @@ const BuiltInProviderNamespace = "builtin"
...
 		panic("called String on zero-value addrs.Provider")
 	}
 	return pt.Hostname.ForDisplay() + "/" + pt.Namespace + "/" + pt.Type
 }
 
+// ForDisplay returns a user-friendly FQN string, simplified for readability. If
+// the provider is using the default hostname, the hostname is omitted.
+func (pt Provider) ForDisplay() string {
+	if pt.IsZero() {
+		panic("called Display on zero-value addrs.Provider")
```
Comment (apparentlymart) [inline addrs/provider.go:59]: This panic message seems to be lagging behind in the renaming. :wink:

**Pull request merged**